src/package-linker.js

```diff
@@ -179,7 +179,7 @@ export default class PackageLinker {
 
       // fs copy can't copy through a symlink, so we replace those with real paths to workpsaces
       for (const [symlink, realpath] of symlinkPaths.entries()) {
-        if (dest !== symlink && dest.indexOf(symlink) === 0) {
+        if (dest.indexOf(symlink + path.sep) === 0) {
```

Comment:
I think the code was misleading, it was not a symlink check.

What I am trying to do here is to replace paths
`node_modules/jest/node_modules/left-pad` with `packages/jest/node_modules/left-pad` because the former is a symlink and we need a real path.

With clause `dest !== symlink` I just made sure I don't change the path of symlink itself.